apps/labs/posts/moving-scipy-to-meson.md

```diff
@@ -0,0 +1,497 @@
+`setuptools` maintainers still plan to do, with a cleaned up API) and then
+update `numpy.distutils` for that change. That would also require moving
+parts of `numpy.distutils` into `setuptools`, like Fortran support (see
+[setuptools/issues/2372](https://github.com/pypa/setuptools/issues/2372) for
+more details). It has become clear though that this will be a really slow and
+painful process - after almost a year, the vendored `distutils` still hasn't
+been re-enabled inside `setuptools`.
+
+The other driver was that SciPy development has become more painful over
+time. SciPy contains C, C++, Fortran and Cython code, exports C and Cython
+APIs, has complex dependencies (see diagram below) and does a lot of code
+generation. The growing amount of C++ and Cython code has increased build
+times. CI systems frequently time out because build plus test suite run takes
+close to 60 minutes (the limit on several free-for-open-source CI offerings).
+Working on compiled code is cumbersome and slow, to the point where a couple
+of other maintainers have told me it's a blocker for them to work on new
+features. Finally, debugging build issues is quite difficult - `distutils`
+doesn't really have a design, so every other extension just monkey patches
+things as needed. Earlier in the year I spent half a day hunting a build
+problem through four (!) different projects that all modified how extensions
+are built.
+
+Those things together made me realize that it's time to move to a better
+build system - if we are forced to do a lot of work because of PEP 632
+anyway, then now is the time. That left the question: which build system?
+Really there were only two viable candidates: CMake and Meson. After some
+experimentation, I had a strong preference for Meson for two main reasons: it
+has *great* [documentation](https://mesonbuild.com/) (unlike CMake), and it's
+easy to contribute to (it is ~25,000 lines of clean Python code, compared to
+~1,000,000 lines of C/C++ for CMake). Given that there are very few Python
+projects using either CMake or Meson, good docs and an easy to understand
+code base are very important - we're going to have to contribute! CMake has a
+larger user base, as well as better Python integration than Meson via
+[scikit-build](https://scikit-build.readthedocs.io/) right now. But to me
+that was less important. Also, scikit-build still depends on `setuptools` and
+one of the goals of this exercise is to get away from distutils/setuptools
+completely and use only a modern build system and Python packaging standards
+(i.e. `pyproject.toml`-based builds) to interact with `pip` and other Python
+packaging tools.
+
+So in February I wrote
+[an RFC titled "switch to Meson as a build system"](https://github.com/scipy/scipy/issues/13615),
+and after positive feedback started working on this project.
+
+<img src="../public/posts/moving-scipy-to-meson/scipy_build_dependencies.png" alt="Diagram of SciPy's build and runtime dependencies" width="600">
```

Comment:
```suggestion
<img src="../public/posts/moving-scipy-to-meson/scipy_build_dependencies.png" alt="Diagram of SciPy's build and runtime dependencies" width="600" />
```

Reply:
@noatamir here's an example of an unclosed `<img>` tag that our markdown compiler doesn't like.